dashboard/app/models/user.rb

```diff
@@ -769,13 +769,39 @@ def track_script_progress(script)
     end
   end
 
-  # returns whether a new level has been completed
-  def track_level_progress(script_level, new_result)
+  # returns whether a new level has been completed and asynchronously enqueues an operation
+  # to update the level progress.
+  def track_level_progress_async(script_level, new_result)
+    new_level_completed = false
```

Comment:
seems like this variable doesn't need to be assigned at the beginning of the method anymore (it could be eliminated and simply replaced with the statement itself as the last line return, or assigned in one statement at the end if you want to keep the meaningful name)